### PR TITLE
Testing the ddCOSMO code

### PR DESCRIPTION
**Pull request by robertodr**
This PR aims to add proper testing for the ddCOSMO code.

## Description
Rework the example into a test and provide the infrastructure necessary for automated testing.

## Todos

* **Developer Interest**

  - [x] Use CTest for testing
  - [x] Rework example into a test using [runtest](https://github.com/bast/runtest)
  - [x] Add support for Travis CI. PRs will from now on be checked by continuous integration.
  - [x] Use Fortran 90 version of [Lebedev-Laikov grid generation](https://people.sc.fsu.edu/~jburkardt/f_src/sphere_lebedev_rule/sphere_lebedev_rule.html)
  - [x] Provide compiler flags also for PGI compilers.
     * Compiler flags are set through the function `set_compiler_flag` that checks that the flags are actually available and working for the current compiler.

## Questions

- [x] @bast I think the way I hooked up with runtest is clumsy, but correct. Could you have a look?

## Status

**IMPORTANT** Before merging change the status badge in `README.md` to `[Build Status](https://travis-ci.org/PCMSolver/ddPCM)` to get the correct status badge.
- [x]  Ready to go

### CODE REVIEW

File: tests/ddCOSMO_Fortran/test.py [hunk 1]
```diff
@@ -0,0 +1,31 @@
...
+# we import essential functions from the runtest library
+from runtest import version_info, get_filter, cli, run
+
+# this tells runtest how to run your code
+from runtest_config import configure
+
+# we stop the script if the major version is not compatible
+assert version_info.major == 2
+
+# construct a filter list which contains two filters
```
Comment (bast) [inline tests/ddCOSMO_Fortran/test.py]: In this case one filter, not two.

**Comment by bast** [comment]
What part about the runtest integration feels clumsy? I had a quick look but wanted to know what feels not completely right to you.

File: tests/ddCOSMO_Fortran/test.py (hunk 1 above)
Comment (robertodr) [reply tests/ddCOSMO_Fortran/test.py]: I updated the comment.

**Comment by robertodr** [comment]
I guess it felt clumsy because there is no launcher script to wrap in `runtest_config.py`, thus the returned tuple contains `full_command` twice.

**Comment by bast** [comment]
Aha - I see, I overlooked that. I will check what we can do about it.

**Review by ilfreddy — APPROVED** [review]

**Review by arnfinn — APPROVED** [review]

**Comment by robertodr** [comment]
The compiler flags selection mechanism needs a bit more testing on Intel and PGI. Opening an issue for it.

**Pull request merged**